### DISCUSSION TITLE
Scaffolding of an existing SQLite database

### DISCUSSION THREAD

**Discussion by DominicPicard** [fnc12/sqlite_orm#1185] [Q&A]
Hi,
I was curious to know if there was a way to "reverse engineer" an existing SQLite database into the make_storage(...) function?
For instance, with EntityFramework it is possible to create a rough DbContext from an existing database.
Thank you!

**Comment by fnc12** [comment]
Hi. This was discussed a lot of times and the idea is very good. One day we need to start. Show your SQL please I'll think how to make it quick. Thanks

**Reply by DominicPicard** [reply]
Hi,
I can't share the database since it is a database that I use at work. But it is good to know that it is something that you discussed.

At the moment I create the `make_storage(...)` function and the associated structs from a little script that parses the commands `PRAGMA table_list;`, `PRAGMA table_info()`. But I am missing the relationships and constraints between the tables. I guess I could possibly find them from `PRAGMA foreign_key_list`... 

But I am lost with how I will specify the different relationships and constraints :(

Thank you
